### DISCUSSION TITLE
How do I refresh a token?

### DISCUSSION THREAD

**Discussion by lassehjorthmadsen** [bluesky-social/atproto#2099] [How to do it?] [answered]
Using [this](https://github.com/bluesky-social/atproto/blob/main/lexicons/com/atproto/server/createSession.json) piece of json specification, I wrote a POST request (supplying an identifier and a password), and succesfully got a token and a refresh token return. 
However, using [this](https://github.com/bluesky-social/atproto/blob/main/lexicons/com/atproto/server/refreshSession.json) specication, I'm not sure how to write a request that **refreshes** my token. It seems that the `input` field is missing, so I don't know what to POST?
I must be missing something, but what?
Thanks a lot for any input.

**Comment by snarfed** [comment, marked as answer]
The lack of `input` can definitely be a surprise! When you POST to `refreshSession`, you include the refresh token you got from `createSession` in the `Authorization: Bearer ...` header, instead of the access token like with other calls.

**Reply by lassehjorthmadsen** [reply]
Many thanks, works like a charm :-) Beautiful piece of work you guys are doing
